### PR TITLE
Utilize `Option::cloned` over explicit `clone` usage.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/compiletest/compiletest.rs [hunk 1]
```diff
@@ -251,10 +245,7 @@ pub fn run_tests(config: &Config) {
 
 pub fn test_opts(config: &Config) -> test::TestOpts {
     test::TestOpts {
-        filter: match config.filter {
-            None => None,
-            Some(ref filter) => Some(filter.clone()),
-        },
+        filter: config.filter.as_ref().cloned(),
```
Comment (Stebalien) [inline src/compiletest/compiletest.rs]: Why not just `config.filter.clone()`?

**Comment by frewsxcv** [comment]
> Why not just config.filter.clone()?

Thanks for pointing this out. I didn't know that calling `clone` on an `&Option` would result in the inner data getting cloned as well.

Force pushed the commit with that change and also updated `.get(0)` to `.first()`.

**Pull request merged**